docs/components/index.md

```diff
@@ -0,0 +1,315 @@
+
+| Directive | Description |
+| --------- | ----------- |
+| [@using](https://docs.microsoft.com/aspnet/core/mvc/views/razor#using) | Adds the C# `using` directive to the generated view. |
+| [@inherits](https://docs.microsoft.com/aspnet/core/mvc/views/razor#inherits) | Provides full control of the class that the component inherits. |
+| [@inject](https://docs.microsoft.com/aspnet/core/mvc/views/razor#inject) | Enables service injection from the [service container](https://docs.microsoft.com/aspnet/core/fundamentals/dependency-injection). For more information, see [Dependency injection into views](https://docs.microsoft.com/aspnet/core/mvc/views/dependency-injection). |
+| [@functions](https://docs.microsoft.com/aspnet/core/mvc/views/razor#functions) | Adds function-level content to a component. Functions include properties and methods accessible to the component. |
+| [@page](https://docs.microsoft.com/aspnet/core/mvc/razor-pages#razor-pages) | Specifies that the component should handle requests directly. The `@page` directive can be specified with a route and optional parameters. Unlike Razor Pages, the `@page` directive doesn't need to be the first directive at the top of the file. |
+
+## Component classes
+
+Blazor's Razor files (*\*.cshtml*) mix HTML markup and C# in the same file, as shown in the sample `HeadingComponent` component above. Blazor doesn't currently support `partial` component classes. However, the `@inherits` directive can be used to provide a Blazor "code-behind" experience.
+
+The [Components Sample app](https://github.com/aspnet/Docs/tree/master/aspnetcore/client-side/blazor/components/common/samples/) shows how a component can inherit a base class, `BlazorRocksBase`, to provide its properties and methods:
+
+*BlazorRocks.cshtml*:
+
+[!code-cshtml[](common/samples/2.x/ComponentsSample/Pages/BlazorRocks.cshtml?start=1&end=8)]
+
+*BlazorRocksBase.cs*:
+
+[!code-cshtml[](common/samples/2.x/ComponentsSample/Pages/BlazorRocksBase.cs)]
+
+The base class must derive from `BlazorComponent` (or implement `IComponent` directly) in the `Microsoft.AspNetCore.Blazor.Components` namespace for this approach to work.
+
+## Data binding
+
+### One-way databinding
+
+One-way databinding is accomplished by prefixing the `@` symbol to any available property. The following example binds:
+
+* The `HeadingFontStyle` property to the CSS property value for `font-style`.
+* `HeadingText` to the content of the `<h1>` element.
+
+```cshtml
+<h1 style="font-style:@HeadingFontStyle">@HeadingText</h1>
+```
+
+### Two-way databinding
+
+Two-way databinding to both components and DOM elements is accomplished with the `bind` attribute. The following example binds the `ItalicsCheck` property to the checkbox's `checked` attribute:
```

Comment:
It's not bound to the checkbox's `checked` attribute; it's bound to the *checked state* of the checkbox. Toggling a checkbox doesn't change its HTML attributes.